File: 2748. Number of Beautiful Pairs/Solution.py
class Solution:
    def countBeautifulPairs(self, nums: List[int]) -> int:
        n, count = len(nums), 0
        for i in range(n):
            for j in range(i + 1, n):
                first, second = nums[i], nums[j] % 10
                while first // 10 > 0: first //= 10
                if self.gcd(first, second) == 1: count += 1
        return count
    
    def gcd(self, x, y):
        if y > x: return self.gcd(y, x)
        if y == 0: return x
        return self.gcd(y, x % y)

File: 2748. Number of Beautiful Pairs/Solution.cpp
class Solution {
public:
    int countBeautifulPairs(vector<int>& nums) {
        int n = nums.size(), count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int first = nums[i], second = nums[j] % 10;
                while (first / 10 > 0) first /= 10;
                if (gcd(first, second) == 1) count++;
            }
        }
        return count;
    }

    int gcd(int x, int y) {
        if (y > x) return gcd(y, x);
        if (y == 0) return x;
        return gcd(y, x % y);
    }
};
